tests/e2e/framework/multicluster.go

```diff
@@ -52,62 +52,50 @@ func getKubeConfigFromFile(dirname string) (string, error) {
 	return remoteKube, nil
 }
 
-func (k *KubeInfo) generateRemoteIstio(src, dst string) error {
-	content, err := ioutil.ReadFile(src)
-	if err != nil {
-		log.Errorf("cannot read remote yaml file %s", src)
-		return err
-	}
+func (k *KubeInfo) getEndpointIPForService(svc string) (ip string, err error) {
 	getOpt := meta_v1.GetOptions{IncludeUninitialized: true}
-	var statsdEPS, pilotEPS, mixerEPS *v1.Endpoints
-
+	var eps *v1.Endpoints
+	// Wait until endpoint is obtained
 	for i := 0; i <= 200; i++ {
-		pilotEPS, err = k.KubeClient.CoreV1().Endpoints(k.Namespace).Get("istio-pilot", getOpt)
-		if (len(pilotEPS.Subsets) != 0) && (err == nil) {
-			mixerEPS, err = k.KubeClient.CoreV1().Endpoints(k.Namespace).Get("istio-policy", getOpt)
-			if (len(mixerEPS.Subsets) != 0) && (err == nil) {
-				statsdEPS, err = k.KubeClient.CoreV1().Endpoints(k.Namespace).Get("istio-statsd-prom-bridge", getOpt)
-				if (len(statsdEPS.Subsets) != 0) && (err == nil) {
-					break
-				}
-			}
+		eps, err = k.KubeClient.CoreV1().Endpoints(k.Namespace).Get(svc, getOpt)
+		if (len(eps.Subsets) == 0) || (err != nil) {
+			time.Sleep(time.Second * 1)
+		} else {
+			break
 		}
-		time.Sleep(time.Second * 1)
-	}
-	if err != nil {
```

Comment:
do you no longer have to check for err (or loop reaching "i=200" )and print an error message?

Reply:
the logic in line 68 should get  this.